src/update-feeds-loops/get-updatable-feeds.ts

```diff
@@ -19,122 +19,152 @@ export interface UpdatableBeacon {
-        //    to change. See:
-        //    https://github.com/api3dao/airnode-protocol-v1/blob/65a77cdc23dc5434e143357a506327b9f0ccb7ef/contracts/api3-server-v1/DataFeedServer.sol#L120
-        // 2. Data feed is a beacon set - We need make sure that the beacon set will change. The contract requires the
-        //    beacon set value or timestamp to change. See:
-        //    https://github.com/api3dao/airnode-protocol-v1/blob/65a77cdc23dc5434e143357a506327b9f0ccb7ef/contracts/api3-server-v1/DataFeedServer.sol#L54
-        const isBeaconSet = beaconValues.length > 1;
-        // Note, that the beacon set value/timestamp is computed as median, so single beacon update may not result in a beacon set update.
-        if (isBeaconSet && newDataFeedValue === dataFeedValue && newDataFeedTimestamp === dataFeedTimestamp) {
-          return false;
-        }
-        if (!isBeaconSet && newDataFeedTimestamp === dataFeedTimestamp) return false;
-
-        return logger.runWithContext(
-          {
-            dapiName: decodedDapiName,
-            dataFeedId,
-          },
-          () =>
-            checkUpdateCondition(
-              dataFeedValue,
-              dataFeedTimestamp,
-              newDataFeedValue,
-              newDataFeedTimestamp,
-              adjustedHeartbeatInterval,
-              adjustedDeviationThresholdCoefficient,
-              deviationReference
-            )
-        );
-      })
-      // Compute the updateable beacons.
-      .map(({ dataFeedInfo, aggregatedBeaconsWithData }) => ({
+        },
+        () =>
+          checkUpdateCondition(
+            dataFeedValue,
+            dataFeedTimestamp,
+            newDataFeedValue,
+            newDataFeedTimestamp,
+            adjustedHeartbeatInterval,
+            adjustedDeviationThresholdCoefficient,
+            deviationReference
+          )
+      );
+
+    if (dataFeedNeedsUpdate) {
+      acc.push({
         dataFeedInfo,
         updatableBeacons: aggregatedBeaconsWithData
           .filter(({ isUpdatable }) => isUpdatable)
           .map(({ beaconId, signedData }) => ({
             beaconId,
             signedData: signedData!,
           })),
-      }))
-  );
+        shouldUpdateBeaconSet: isBeaconSet,
+      });
+    } else if (isBeaconSet && asyncBeaconUpdatedeviationThresholdFactor) {
+      // 2.5. There is a special case when data feed is a beacon set that do not need to be updated but some of its beacon constituents do.
+      //      In this particular case, airseeker can update only these beacons and skip the beacon set update. This is enabled by setting a
+      //      value in asyncBeaconUpdatedeviationThresholdFactor on the config.
+      const updatableBeacons = aggregatedBeaconsWithData
+        .filter(
+          ({ isUpdatable, offChainValue, onChainValue }) =>
+            isUpdatable &&
+            offChainValue &&
+            checkUpdateCondition(
```

Comment:
Maybe it's better to wrap this call with `logger.runWithContext` like above.

Reply:
https://github.com/api3dao/airseeker/pull/343/commits/4089d86ea8283c4b34042d57ff37f8616b708c09